### PR TITLE
#297 skeleton.xsd:31-33: Skeleton.xsd: remove method/@public.... 

### PR DESCRIPTION
**Pull request by GnusinPavel**
#297 attribute public removed from method tag

### CODE REVIEW

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#346](https://github.com/yegor256/jpeek/issues/346) is now in scope, role is `REV`

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) everybody who has role `REV` is banned at [#346](https://github.com/yegor256/jpeek/issues/346); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by GnusinPavel** [comment]
@paulodamaso can you explain me, why I deleted few lines lines of code, wrote a new test, but coverage goes down?

**Comment by 0crat** [comment]
This pull request [#346](https://github.com/yegor256/jpeek/issues/346) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/5d58e8d9-bd6f-4bb0-95e5-394b31b7fdd5); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by GnusinPavel** [comment]
@fabriciofx can you review this pull request?

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java [hunk 1]
```diff
@@ -65,6 +66,18 @@ public void parsesMethodVisibility() {
         );
     }
 
+    @Test
+    public void thereIsNoAttributePublic() {
+        MatcherAssert.assertThat(
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/XmlClassTest.java]: @GnusinPavel please use `Assertion` (from `cactoos-matchers`) here.

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java [hunk 2]
```diff
@@ -65,6 +66,18 @@ public void parsesMethodVisibility() {
...
+            XhtmlMatchers.xhtml(
+                this.classAsXml("ClassWithDifferentMethodVisibilities")
+            ),
+            XhtmlMatchers.hasXPaths(
+                "/class/methods/method[not (@public)]"
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/XmlClassTest.java]: @GnusinPavel and the tests for the others visibility attributes?

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java [hunk 3]
```diff
@@ -65,6 +66,18 @@ public void parsesMethodVisibility() {
         );
     }
 
+    @Test
+    public void thereIsNoAttributePublic() {
+        MatcherAssert.assertThat(
+            XhtmlMatchers.xhtml(
+                this.classAsXml("ClassWithDifferentMethodVisibilities")
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/XmlClassTest.java]: @GnusinPavel please, leave a puzzle to extract the method `classAsXml` as a new class. @paulodamaso WDYT?

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java (hunk 2 above)
Comment (GnusinPavel) [reply src/test/java/org/jpeek/skeleton/XmlClassTest.java]: there is one already, https://github.com/yegor256/jpeek/blob/d0e9c60d253466d1e41fc79a7735777a42385745/src/test/java/org/jpeek/skeleton/XmlClassTest.java#L54-L67

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java (hunk 1 above)
Comment (GnusinPavel) [reply src/test/java/org/jpeek/skeleton/XmlClassTest.java]: replaced

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java (hunk 3 above)
Comment (GnusinPavel) [reply src/test/java/org/jpeek/skeleton/XmlClassTest.java]: put puzzle

**Comment by 0crat** [comment]
@fabriciofx[/z](https://www.0crat.com/u/fabriciofx) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
Job [#346](https://github.com/yegor256/jpeek/issues/346) is already in the agenda of @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Pull request [#346](https://github.com/yegor256/jpeek/issues/346) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by 0crat** [comment]
The job [#346](https://github.com/yegor256/jpeek/issues/346) is now out of scope

**Comment by 0crat** [comment]
Code review was too long (36 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by fabriciofx** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/jpeek/issues/346#issuecomment-554653294))

@fabriciofx This is what I know about this job in [C7JGJ00DP](https://www.0crat.com/p/C7JGJ00DP), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#346](https://github.com/yegor256/jpeek/issues/346) is not in scope
  * The job is waiting quality review verdict by @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) for 5days
  * @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C7JGJ00DP?q=%7Bjob%3A%27gh%3Ayegor256%2Fjpeek%23346%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by fabriciofx** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Job [#346](https://github.com/yegor256/jpeek/issues/346) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#346](https://github.com/yegor256/jpeek/issues/346) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/cb42a4c9-d6a6-49f1-8e9b-879710596c79); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by fabriciofx** [comment]
@GnusinPavel it seems fine to me. @paulodamaso can you merge it, please?

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by GnusinPavel** [comment]
@paulodamaso fixed

**Comment by fabriciofx** [comment]
> > @rultor merge
> 
> @GnusinPavel @paulodamaso Oops, I failed. You can see the full log [here](https://www.rultor.com/t/20896-555179353) (spent 13min)
> 
> ```
> Reading src/main/java/org/jpeek/Report.java...
> Puzzle 227-fbf042c2 30/DEV at src/main/java/org/jpeek/Report.java
> Reading src/main/java/org/jpeek/DefaultBase.java...
> Reading src/main/java/org/jpeek/web/TkQueue.java...
> Reading src/main/java/org/jpeek/web/Sigmas.java...
> Reading src/main/java/org/jpeek/web/TkReport.java...
> Reading src/main/java/org/jpeek/web/RsPage.java...
> Reading src/main/java/org/jpeek/web/Futures.java...
> Reading src/main/java/org/jpeek/web/Pages.java...
> Reading src/main/java/org/jpeek/web/TypedPages.java...
> Reading src/main/java/org/jpeek/web/Results.java...
> Reading src/main/java/org/jpeek/web/Dynamo.java...
> Reading src/main/java/org/jpeek/web/package-info.java...
> Reading src/main/java/org/jpeek/web/Front.java...
> Reading src/main/java/org/jpeek/web/Mistakes.java...
> Reading src/main/java/org/jpeek/web/TkMistakes.java...
> Reading src/main/java/org/jpeek/web/AsyncReports.java...
> Reading src/main/java/org/jpeek/web/TkIndex.java...
> Reading src/main/java/org/jpeek/web/DyNum.java...
> Reading src/main/java/org/jpeek/web/StickyFutures.java...
> Reading src/main/java/org/jpeek/web/TkApp.java...
> Reading src/main/java/org/jpeek/web/Reports.java...
> Reading src/main/java/org/jpeek/web/TkAll.java...
> Reading Procfile...
> Reading deploy.sh...
> Reading pom.xml...
> Reading CHECKS...
> Reading research/2018-March/03-filter-out-artifacts/target-926.csv...
> Reading research/2018-March/05-filter-out-non-normal/run.sh...
> Reading research/2018-March/05-filter-out-non-normal/.gitignore...
> Reading research/2018-March/05-filter-out-non-normal/filter.php...
> Reading research/2018-March/06-join-and-add-ranks/run.sh...
> Reading research/2018-March/06-join-and-add-ranks/join-and-rank.php...
> Reading research/2018-March/06-join-and-add-ranks/.gitignore...
> Reading research/2018-March/02-count-classes/run.sh...
> Reading research/2018-March/02-count-classes/classes-in-artifact.sh...
> Reading research/2018-March/02-count-classes/all-with-counts.csv...
> Reading research/2018-March/07-draw-graph/run.sh...
> Reading research/2018-March/07-draw-graph/.gitignore...
> Reading research/2018-March/07-draw-graph/draw.php...
> Reading research/2018-March/01-find-all-artifacts/2017-full-list.txt...
> Reading research/2018-March/04-calculate-mu-and-sigma/parse-index.php...
> Reading research/2018-March/04-calculate-mu-and-sigma/run.sh...
> Reading research/2018-March/04-calculate-mu-and-sigma/get-sigma-and-mu.sh...
> Reading research/2018-March/04-calculate-mu-and-sigma/.gitignore...
> Reading research/2018-April/03-filter-out-artifacts/target-926.csv...
> Reading research/2018-April/04-collect-metrics/process-artifact.sh...
> Reading research/2018-April/04-collect-metrics/run.sh...
> Reading research/2018-April/04-collect-metrics/collect.rb...
> Reading research/2018-April/04-collect-metrics/.gitignore...
> Reading research/2018-April/05-draw-graphs/run.sh...
> Reading research/2018-April/05-draw-graphs/draw.rb...
> Reading research/2018-April/05-draw-graphs/.gitignore...
> Reading .gitattributes...
> \u001b[31mERROR\u001b[0m: Puzzle src/test/java/org/jpeek/skeleton/XmlClassTest.java:83-83 has a very short description of just 7 words while a minimum of 20 is required
> \u001b[31mERROR\u001b[0m: 1 errors, see log above
> If you can't understand the cause of this issue or you don't know how to fix it, please submit a GitHub issue, we will try to help you: https://github.com/yegor256/pdd/issues. This tool is still in its beta version and we will appreciate your feedback. Here is where you can find more documentation: https://github.com/yegor256/pdd/blob/master/README.md.
> Exit code is 1
> container ba604cc0d7ca770b15a86460ff684630aa0dfc6220dee55290d4e7ed81b09da7 is dead
> Mon Nov 18 21:03:21 CET 2019
> ```

I think CI can't catch this kind of error a bug. Should I open an issue about it?

**Comment by GnusinPavel** [comment]
@paulodamaso once again )

**Comment by fabriciofx** [comment]
@paulodamaso ping

**Comment by fabriciofx** [comment]
@paulodamaso ping

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#346](https://github.com/yegor256/jpeek/issues/346) is already in the agenda of @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@aadca18db847, thread: PQ-C7JGJ00DP
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 1 error(s) in XML document: -1:-1: Duplicate unique value [gh:yegor256/jpeek#346] declared for identity constraint of element "speed".
```

[0.50.36](https://github.com/zerocracy/farm): CID: [0e4eeb25-a93e-4bd1-a27a-aebebdffd9a3](https://www.0crat.com/footprint/C7JGJ00DP/0e4eeb25-a93e-4bd1-a27a-aebebdffd9a3), Type: "Order was finished"

**Comment by 0crat** [comment]
The job [#346](https://github.com/yegor256/jpeek/issues/346) is now out of scope

**Comment by 0crat** [comment]
Code review was too long (8 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by fabriciofx** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Pull request merged**